packages/ember-data/lib/system/store.js

```diff
@@ -331,6 +331,10 @@ Store = Service.extend({
     // Set the properties specified on the record.
     record.setProperties(properties);
 
+    record.eachRelationship(function(key, descriptor) {
+      record._relationships[key].setHasData(true);
```

Comment:
The `true` argument is not used within the `relationship`s `setHasData` method ...